### PR TITLE
Export internal headers

### PR DESCRIPTION
**Pull request by ssd04**

### CODE REVIEW

**Review by bogdan-rosianu — CHANGES_REQUESTED** [review]

File: process/blockProcessor_test.go [hunk 1]
```diff
@@ -392,3 +395,445 @@ func TestBlockProcessor_GetHyperBlock(t *testing.T) {
...
+	require.Equal(t, process.ErrSendingRequest, err)
+	require.Nil(t, res)
+}
+
+func TestBlockProcessor_GetInternalBlockByNonceShouldWork(t *testing.T) {
+	t.Parallel()
+
+	ts := &testStruct{
+		Nonce: 10000,
+		Name:  "a test struct to be send",
```
[the remark this answers is not in the document]
Comment (ssd04) [reply process/blockProcessor_test.go]: :+1:

File: process/errors.go [hunk 2]
```diff
@@ -100,3 +100,6 @@ var ErrNilLogsMerger = errors.New("nil logs merger")
 
 // ErrNilSCQueryService signals that a nil smart contracts query service has been provided
 var ErrNilSCQueryService = errors.New("nil smart contracts query service provided")
+
+// ErrInvalidOutportFormat signals that the outport format type is not valid
+var ErrInvalidOutportFormat = errors.New("the outport format type is invalid")
```
[the remark this answers is not in the document]
Comment (ssd04) [reply process/errors.go]: :+1:

File: api/groups/baseInternalGroup.go [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+			c,
+			http.StatusBadRequest,
+			nil,
+			apiErrors.ErrInvalidBlockHashParam.Error(),
+			data.ReturnCodeRequestError,
+		)
+		return
+	}
+
+	blockByHashResponse, err := group.facade.GetInternalMiniBlockByHash(shardID, hash, common.Internal)
```
[the remark this answers is not in the document]
Comment (ssd04) [reply api/groups/baseInternalGroup.go]: changed with miniBlock prefix

File: api/groups/baseInternalGroup.go [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+			c,
+			http.StatusBadRequest,
+			nil,
+			apiErrors.ErrInvalidBlockHashParam.Error(),
+			data.ReturnCodeRequestError,
+		)
+		return
+	}
+
+	blockByHashResponse, err := group.facade.GetInternalMiniBlockByHash(shardID, hash, common.Proto)
```
[the remark this answers is not in the document]
Comment (ssd04) [reply api/groups/baseInternalGroup.go]: changed with miniBlock prefix

File: data/block.go [hunk 5]
```diff
@@ -111,3 +111,27 @@ type EpochStartInfo struct {
...
+// InternalMiniBlockApiResponse is a response holding an internal miniblock
+type InternalMiniBlockApiResponse struct {
+	Data  InternalMiniBlockApiResponsePayload `json:"data"`
+	Error string                              `json:"error"`
+	Code  ReturnCode                          `json:"code"`
+}
+
+// InternalMiniBlockApiResponsePayload wraps an internal miniblock
+type InternalMiniBlockApiResponsePayload struct {
+	Block interface{} `json:"miniblock"`
```
[the remark this answers is not in the document]
Comment (ssd04) [reply data/block.go]: done

File: process/blockProcessor_test.go [hunk 6]
```diff
@@ -392,3 +395,445 @@ func TestBlockProcessor_GetHyperBlock(t *testing.T) {
 	require.Equal(t, 42, int(response.Data.Hyperblock.Nonce))
 	require.Equal(t, "abcd", response.Data.Hyperblock.Hash)
 }
+
+// GetInternalBlockByNonce
+
+func TestBlockProcessor_GetInternalBlockByNonceInvalidOutportFormat_ShouldFail(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (ssd04) [reply process/blockProcessor_test.go]: renaming done

File: process/blockProcessor.go [hunk 7]
```diff
@@ -156,3 +170,141 @@ func (bp *BlockProcessor) GetHyperBlockByNonce(nonce uint64) (*data.HyperblockAp
...
+
+		log.Info("miniblock request", "shard id", observer.ShardId, "hash", hash, "observer", observer.Address)
+		return &response, nil
+
+	}
+
+	return nil, ErrSendingRequest
+}
+
+func getOutportStr(format common.OutportFormat) (string, error) {
```
[the remark this answers is not in the document]
Comment (ssd04) [reply process/blockProcessor.go]: done

File: common/constants.go [hunk 8]
```diff
@@ -0,0 +1,12 @@
+package common
+
+// OutportFormat represents the format type returned by api
+type OutportFormat uint8
```
[the remark this answers is not in the document]
Comment (ssd04) [reply common/constants.go]: :+1:

**Review by bogdan-rosianu — DISMISSED** [review]

**Review by AdoAdoAdo — DISMISSED** [review]

**Review by AdoAdoAdo — APPROVED** [review]

**Pull request merged**